ssl/ssl_sess.c

```diff
@@ -26,54 +26,37 @@ static int remove_session_lock(SSL_CTX *ctx, SSL_SESSION *c, int lck);
 
 DEFINE_STACK_OF(SSL_SESSION)
 
-__owur static int sess_timedout(time_t t, SSL_SESSION *ss)
+__owur static ossl_inline OSSL_TIME time_t_to_ossl_time(time_t t)
 {
-    /* if timeout overflowed, it can never timeout! */
-    if (ss->timeout_ovf)
-        return 0;
-    return t > ss->calc_timeout;
+    return t * (OSSL_TIME_SECOND / 1000000);
```

Comment:
I don't believe this conversion is guaranteed to be correct by the standards.

Reply:
Yeah, I think I conflated a few things copying from `ossl_time_now()`. I was really hoping @paulidale would introduce a proper equivalent!

Reply:
Been on vacation, so couldn't add anything.  It looks like you've added the conversion function.

Reply:
It looks like we assume the existence of `struct timeval`. The very definition of that struct assumes that a `time_t` is a value in seconds:

````
time_t         tv_sec      seconds
````

Since we assume that `struct timeval` exists I guess we can rely on this conversion.

Reply:
Coverity has bought up a number of "errors" with our time_t usage.  There might be some rework required.